test/libsolidity/natspecJSON/custom_inheritance.sol

```diff
@@ -0,0 +1,43 @@
+contract A {
+    /// @custom:since 2014
+    function g(uint x) public pure virtual {}
+}
+contract B is A {
+    function g(uint x) public pure override {}
+}
+
+// ----
+// ----
+// :A devdoc
+// {
+//     "kind": "dev",
+//     "methods":
+//     {
+//         "g(uint256)":
+//         {
+//             "custom:since": "2014"
+//         }
+//     },
+//     "version": 1
+// }
+//
+// :A userdoc
+// {
+//     "kind": "user",
+//     "methods": {},
+//     "version": 1
```

Comment:
By the way, we could filter out `kind` and `version` to make the expectations a little shorter. Should we? `version` is the same for all tests and `kind` is redundant.

The original tests were inconsistent about it. Some had these fields, but many did not. The test case was also ignoring them, which meant they were not checked (I actually find one or two places where we were expecting the wrong value), but the potential for these to be wrong is pretty low.

Reply:
I'd keep them.